src/main/java/sixgaezzang/sidepeek/skill/serivce/SkillService.java

```diff
@@ -0,0 +1,42 @@
+package sixgaezzang.sidepeek.skill.serivce;
+
+import static sixgaezzang.sidepeek.common.ValidationUtils.validateMaxLength;
+
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import sixgaezzang.sidepeek.skill.domain.Skill;
+import sixgaezzang.sidepeek.skill.dto.SkillResponse;
+import sixgaezzang.sidepeek.skill.dto.SkillSearchResponse;
+import sixgaezzang.sidepeek.skill.repository.SkillRepository;
+
+@Service
+@RequiredArgsConstructor
+public class SkillService {
+
+    public static final int KEYWORD_MAX_LENGTH = 50;
+
+    private final SkillRepository skillRepository;
+
+    public SkillSearchResponse searchByName(String keyword) {
+        List<Skill> skills;
+        if (Objects.isNull(keyword) || keyword.isBlank()) {
+            skills = skillRepository.findAll();
+        } else {
+            validateMaxLength(keyword, KEYWORD_MAX_LENGTH,
+                "최대 " + KEYWORD_MAX_LENGTH + "자의 키워드로 검색할 수 있습니다.");
+            skills = skillRepository.findAllByNameContaining(keyword);
+        }
+
+        List<SkillResponse> searchResults = skills.stream()
+            .map(skill -> SkillResponse.builder()
+                .id(skill.getId())
+                .name(skill.getName())
+                .iconImageUrl(skill.getIconImageUrl())
+                .build()
+            ).toList();
+
+        return new SkillSearchResponse(searchResults);
```

Comment:
```suggestion
        return SkillSearchResponse.from(searchResults);
```

`SkillSearchResponse` 에서도 new 보다는 정적 팩토리 메서드를 활용하면 다음과 같이 간단해질 수 있을 것 같습니다! 혹시 세희님은 이에 대해 어떻게 생각하시는지 궁금합니다!

Reply:
저도 적극 찬성합니다! 적극반영하겠습니당!!